src/ipelib/dynamo/module_weimer2005Ipe.f

```diff
@@ -0,0 +1,806 @@
+      endif
+      end subroutine setmodel2005Ipe
+!-----------------------------------------------------------------------
+      subroutine setboundary2005Ipe(angle,bt,tilt,swvel,swden,file_path &
+     & )
+      implicit none
+!
+! Args:
+!
+! file_path: directory in which to find data file (must have "/" at end)
+      character(len=*),intent(in) :: file_path
+      real*8,intent(in) :: angle,bt,tilt,swvel,swden
+!
+! Local:
+      integer :: i
+      real*8 :: swp,xc,theta,ct,st,tilt2,cosa,btx,x(na),c(na)
+
+!   write(6,"('Enter setboundary: angle=',f8.3,' bt=',f8.3)") angle,bt
+!
+! Read data:
+      call read_bndy2005Ipe('global_idea_coeff_W05scBndy.dat')
+!
+! Calculate the transformation matrix to the coordinate system
+! of the offset pole.
+!
+      xc = 4.2
+      theta = xc*(deg2rad)
+      ct = cos(theta)
+      st = sin(theta)
+!
+      tmat(1,:) = (/ ct, 0.d0, st/) 
+      tmat(2,:) = (/ 0.d0, 1.d0, 0.d0/) 
+      tmat(3,:) = (/-st, 0.d0, ct/)
+!     
+      ttmat(1,:) = (/ct, 0.d0,-st/)
+      ttmat(2,:) = (/ 0.d0,1.d0, 0.d0/)
+      ttmat(3,:) = (/st, 0.d0, ct/)
+!
+      swp = swden*swvel**2*1.6726d-6 ! pressure
+      tilt2 = tilt**2
+      cosa = cos(angle*deg2rad)
+      btx = 1.-exp(-bt*ex_bndy(1))
+      if (bt > 1.) then
+         btx = btx*bt**ex_bndy(2)
+      else
+         cosa = 1.+bt*(cosa-1.) ! remove angle dependency for IMF under 1 nT
+      endif
+      x = (/1.d0, cosa, btx, btx*cosa, swvel, swp/)
+      c = bndya
+      bndyfitr = 0.
+      do i=1,na
+         bndyfitr = bndyfitr+x(i)*c(i)
+      enddo
+
+!   write(6,"('setboundary: cosa=',f8.3,' btx=',f8.3)") cosa,btx
+!   write(6,"('setboundary: x=',/,(6e12.4))") x
+!   write(6,"('setboundary: c=',/,(6e12.4))") c
+!   write(6,"('setboundary: bndyfitr=',e12.4)") bndyfitr
+
+      end subroutine setboundary2005Ipe
+!-----------------------------------------------------------------------
+      subroutine epotval2005Ipe(lat,mlt,fill,epot)
+      implicit none
+!
+! Args:
+      real*8,intent(in) :: lat,mlt,fill
+      real*8,intent(out) :: epot
+!     
+! Local:
+      integer :: inside,j,m,mm,skip
+      real*8 :: z,phir,plm,colat,nlm
+      real*8 :: phim(2),cospm(2),sinpm(2)
+!
+! checkinputs returns inside=1 if lat is inside model boundary,
+! inside=0 otherwise. Phir and colat are also returned by checkinputs.
+!     
+      call checkinputs2005Ipe(lat,mlt,inside,phir,colat)
+      if (inside == 0) then
+         epot = fill
+         return
+      endif
+!
+! IDL code: 
+! phim=phir # replicate(1,maxm) * ((indgen(maxm)+1) ## replicate(1,n_elements(phir)))
+!   where the '#' operator multiplies columns of first array by rows of second array,
+!   and the '##' operator multiplies rows of first array by columns of second array.
+! Here, maxm == maxm_pot == 2 (from w05read_data module), and phir is a scalar. The 
+!   above IDL statement then becomes: phim = ([phir] # [1,1]) * ([1,2] ## [phir]) where
+!   phim will be dimensioned [1,2]
+!
```

Comment:
Please remove test code and comments if validated.

Reply:
I will leave it since it comes with the weimer package.